lib/utils/reader.cpp

```diff
@@ -7,79 +7,101 @@ namespace utils{
 
 Reader::Reader(){}
 
-//this function is probably now useless but still good practise!
-void Reader::readFirstProblem(const std::string problems_filepath){
+void Reader::readNextProblem(std::fstream &problems_filestream){
```

Comment:
Ngl probably gonna change the implementation to construct a reader we can both work with and then give us separate formatters for input to our solvers...